File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py

def solution(cacheSize, cities):
    answer = 0

    caches = []

    for city in cities:
        ct = city.upper()
        if ct in caches:
            answer += 1
            caches.remove(ct)
            caches.append(ct)
        else:
            if len(caches) < cacheSize:
                caches.append(ct)
                answer += 5
            elif cacheSize == 0:
                answer += 5
            else:
                caches = caches[1:]
                caches.append(ct)
                answer += 5

    return answer

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.swift
func solution(_ cacheSize:Int, _ cities:[String]) -> Int {

    var cache: [String] = []
    var result = 0
    
    for city in cities {
        var newCache: [String] = []
        
        var hit = ""
        for ch in cache {
            if ch == city.lowercased() { 
                hit = ch
                continue
            }
            
            newCache.append(ch)
        }
        
        if !hit.isEmpty { 
            newCache.append(hit) 
            result += 1
        } else {
            if cacheSize != 0 {
                if newCache.count >= cacheSize {
                newCache.removeFirst()
                }
                newCache.append(city.lowercased())    
            }
            
            result += 5
        }
        cache = newCache
        
    }
    
    return result
}
